src/library/dao/playlistdao.cpp

```diff
@@ -380,13 +380,96 @@ bool PlaylistDAO::isHidden(const int playlistId) const {
     return true;
 }
 
-void PlaylistDAO::removeTrackFromPlaylist(const int playlistId, const int position)
-{
+
+void PlaylistDAO::removeHiddenTracks(const int playlistId) {
+	ScopedTransaction transaction(m_database);
+    // This query deletes all tracks marked as deleted and all
+    // phantom track_ids with no match in the library table
+    QString queryString = QString(
+            "SELECT position FROM PlaylistTracks "
```

Comment:
Why don't you execute "DELETE FROM PlaylistTracks..." directly? Afterwards you can adjust the position of all remaining tracks in this playlist. Or am I missing something here?

Adjusting the position of playlist tracks should be common operation. You may also provide a start position (which is known to be valid, default = 0) and adjust only the positions greater than this start position.